srcpkgs/bitcoin/template

```diff
@@ -1,39 +1,49 @@
 # Template file for 'bitcoin'
 pkgname=bitcoin
-version=0.19.1
-revision=4
+version=0.21.0
+revision=1
 build_style=gnu-configure
 configure_args="--with-incompatible-bdb --disable-ccache --disable-static
- --enable-hardening --with-boost=${XBPS_CROSS_BASE}/usr"
-hostmakedepends="pkg-config yasm"
-makedepends="db-devel boost-devel miniupnpc-devel
- libevent-devel qrencode-devel libatomic-devel"
+ --enable-hardening --with-boost=${XBPS_CROSS_BASE}/usr --with-gui"
+hostmakedepends="automake libtool pkg-config yasm"
+makedepends="db-devel boost-devel miniupnpc-devel qt5-tools-devel
+ libevent-devel qrencode-devel libatomic-devel sqlite-devel"
 short_desc="Bitcoin is a peer-to-peer network based digital currency"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="MIT"
 homepage="https://bitcoin.org/"
 distfiles="https://bitcoin.org/bin/bitcoin-core-${version}/bitcoin-${version}.tar.gz
- https://raw.github.com/bitcoin-core/packaging/master/debian/bitcoin-qt.desktop
- https://raw.github.com/bitcoin/bitcoin/v${version}/share/pixmaps/bitcoin128.png"
-checksum="f2591d555b8e8c2e1bd780e40d53a91e165d8b3c7e0391ae2d24a0c0f23a7cc0
+ https://raw.githubusercontent.com/bitcoin-core/packaging/${version%.*}/debian/bitcoin-qt.desktop
+ https://raw.githubusercontent.com/bitcoin/bitcoin/v${version}/share/pixmaps/bitcoin128.png"
+checksum="1a91202c62ee49fb64d57a52b8d6d01cd392fffcbef257b573800f9289655f37
  0a46bbadda140599e807be38999e6848c89f9c3523d26fede02d34d62d50f632
  ad880c8459ecfdb96abe6a4689af06bdd27906e0edcd39d0915482f2da91e722"
 conflicts="litecoin>=0" # Both provide libbitcoinconsensus.so.0
 skip_extraction="bitcoin-qt.desktop bitcoin128.png"
-subpackages="bitcoin-devel"
 
-if [ -z "$CROSS_BUILD" ]; then
-	# XXX -qt subpkg
-	makedepends+=" qt5-devel qt5-tools-devel"
-	configure_args+=" --with-gui"
-	subpackages+=" bitcoin-qt"
+if [ "$CROSS_BUILD" ]; then
+	hostmakedepends+=" qt5-host-tools"
 fi
 
+pre_configure() {
+	autoreconf -fi
+	case "$XBPS_TARGET_MACHINE" in
+	aarch64*)
+		CFLAGS=${CFLAGS/armv8-a/armv8-a+crc+crypto}
```

Comment:
Can you explain why this is necessary? Does it fail to build without it?

Reply:
Yes, it does. Simple test:
```c
/* sample.c */
#include <arm_acle.h>

int main()
{
    __crc32cd(0, 0);
    return 0;
}
```
```console
$ aarch64-linux-musl-gcc sample.c
/usr/lib/gcc/aarch64-linux-musl/10.2/include/arm_acle.h:128:1: error: inlining failed in call to 'always_inline' '__crc32cd': target specific option mismatch
  128 | __crc32cd (uint32_t __a, uint64_t __b)
      | ^~~~~~~~~
sample.c:5:5: note: called from here
    5 |     __crc32cd(0, 0);
```
```console
$ aarch64-linux-musl-gcc -march=armv8-a+crc+crypto sample.c ; echo $?
0
```